Update dependencies

silvanshade opened the pull request:
This change updates all dependencies to their latest released versions.

Dependencies updated:

- `io-uring`
- `local-sync`
- `mio`
- `nix`
- `socket2`

This is mainly to reduce duplicates of these in the build graph of downstream crates using newer versions.

EDIT: I removed the updates to `windows-sys` because it requires deeper refactoring to the internals.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/bytedance/monoio?pullRequest=363) <br/>All committers have signed the CLA.

silvanshade committed:
Replace fxhash dep with rustc_hash

This fixes the cargo-deny error due to fxhash being unmaintained:

https://rustsec.org/advisories/RUSTSEC-2025-0057

silvanshade committed:
ci: Skip io_uring arch check for Tier 2 platforms

Related:

- https://github.com/tokio-rs/io-uring/pull/361#discussion_r2314811654
- https://github.com/tokio-rs/io-uring/pull/362

silvanshade committed:
Update dependencies

This change updates all dependencies to their latest released versions.

Dependencies updated:

- io-uring
- local-sync
- mio
- nix
- socket2
